### PR TITLE
feat: inputNumber support status

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/33854
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | InputNumber support customized status|
| 🇨🇳 Chinese |  InputNumber 支持自定义状态|

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input-number/index.tsx [hunk 1]
```diff
@@ -62,6 +66,9 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
       );
   }
 
+  const { hasFeedback, status: contextStatus } = useContext(FormItemStatusContext);
+  const mergedStatus = contextStatus || customStatus;
```
Comment (zombieJ) [inline components/input-number/index.tsx:70]: 这个是不是应该自定义更高一些？比如自定义复合组件可能盖不掉

Comment (MadCcc) [reply components/input-number/index.tsx]: done

File: components/_util/statusUtils.tsx [hunk 2]
```diff
@@ -39,3 +39,11 @@ export function getStatusClassNames(
     [`${prefixCls}-has-feedback`]: hasFeedback,
   });
 }
+
+export const getMergedStatus = ({
+  contextStatus,
+  customStatus,
+}: {
```
Comment (zombieJ) [inline components/_util/statusUtils.tsx:46]: 简单点，直接用入参不用 object 对象。object 的 fieldName 没办法被 uglify

**Review by zombieJ — APPROVED** [review]

File: components/input-number/style/status.less [hunk 3]
```diff
@@ -0,0 +1,45 @@
+@import '../../input/style/status';
```
Comment (afc163) [inline components/input-number/style/status.less:1]: 这里是不是有类似的问题？都查一遍。

Comment (MadCcc) [reply components/input-number/style/status.less:1]: 嗯对，我待会把 input 相关的都改一改

**Pull request merged**